### DISCUSSION TITLE
ReCaptcha is not working?

### DISCUSSION THREAD

**Discussion by PeopleInside** [the-djmaze/snappymail#841] [Q&A] [answered]
I'm trying to activate the recaptcha plugin that I see has been updated 3 weeks ago.
I uploaded the plugin folder in the plugin folder and activated in settings with valid key. Also the domain is inserted, why captcha is not loading?

The dev console say that the loading was blocked from the frame-sc of the page.
![immagine](https://user-images.githubusercontent.com/5006150/211498506-fa650481-262a-4ff1-8047-24292935173c.png)

**Comment by cm-schl** [comment]
Hi,

I'm unsure because I don't use this plugin, but could this maybe have to do with the setting `content_security_policy`?
As said, just an idea. Have a look at: https://github.com/the-djmaze/snappymail/wiki/Admin-Manual#menu-security where at least something about `content_security_policy` was described at the moment...

Otherwise maybe send us some screenshot / log of the console where the blocking of the captcha occurs 🙂

**Reply by PeopleInside** [reply]
Hi, thank you for your reply!
Should not exist any snappymail settings that should block reCaptcha so I dont think is related to the content_security_policy.

To me seems just the reCaptcha plugin doesnt work.
Easy.

I activated it with correct recaptcha V2 and correct codes and is not working.
PHP 8.2

Maybe the developer can see this topic test and reply, I never opened a BUG for this at the moment.
Let's see if some help arrive here.

Plugins can be downloaded and uploaded by FTP or installed directly from the administrative URL.

**Reply by PeopleInside** [reply]
MMmmmMMmm maybe you are not wrong.

I am checking.
https://github.com/the-djmaze/snappymail/issues/222#issuecomment-1036208993

Lets see.

Still not showed also if I add the content security policy

**Comment by the-djmaze** [comment, marked as answer]
Issue is changed domain name as people in China could not access it.
https://github.com/RainLoop/rainloop-webmail/pull/2194

So the CSP fix is here: https://github.com/the-djmaze/snappymail/commit/07dae9dd30e874f7485a94ddadd589854c28066d

**Comment by PeopleInside** [comment]
@the-djmaze I will test. Seems this extension cannot be installed directly in the admin section as is not listed.
Need download and upload by using FTP :)

Testing now if works.
IT works.

Just if the captcha is not selected an unknown error is showed but it ok.
Great thanks!
